lib/node_modules/@stdlib/math/base/special/logaddexp/lib/main.js

```diff
@@ -0,0 +1,64 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2018 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var isnan = require( '@stdlib/math/base/assert/is-nan' );
+var exp = require( '@stdlib/math/base/special/exp' );
+var ln = require( '@stdlib/math/base/special/ln' );
+var log1p = require( '@stdlib/math/base/special/log1p' );
+
+
+// VARIABLES //
+
+var LN2 = ln( 2.0 );
```

Comment:
We use a constant package for `LN2`: `@stdlib/constants/math/float64-ln-two`.

Why? Because `ln(x)` is not guaranteed to remain the same. Meaning, we could change the underlying implementation, which might affect the result of, e.g., L31. Accordingly, to avoid subtle bugs, we rely on constants (exposed as packages) which are truncated (high-precision) values, which should be correctly rounded when converted to double-precision.

Reply:
I believe I made the same mistake with log1mexp too then.

Reply:
I already addressed it in my WIP feature branch. :)